### PR TITLE
Fixes #8272: Removed white space for no title in a StyledProgressDialog dialouge popup

### LINKED ISSUE #8272: Check media and delete deck popup have white space on top
###### Reproduction Steps

1. Check media from options menu or delete deck
2. Popup appear with white space
3. 

###### Expected Result
No white space at top of popup

###### Actual Result
</img>
</img>

###### Debug info
Refer to the [support page](https://ankidroid.org/docs/help.html) if you are unsure where to get the "debug info".

###### Research
*Enter an [x] character to confirm the points below:*

- [ x ] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [ x ] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [ x ] I have searched for similar existing issues here and on the user forum
- [ x ] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by Ekagra**
## Purpose / Description
Removing white space for no title in a StyledProgressDialog dialouge popup

## Fixes
Fixes https://github.com/ankidroid/Anki-Android/issues/8272

## Approach
I looked over every method using `StyledProgressDialog`  with title being "" and replaced with null

## How Has This Been Tested?
<p float="left">
  
</p>

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [x] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Review by david-allison-1 — COMMENTED** [review]
Thanks!!

Can we fix this in `StyledProgressDialog.show` as well - translate `""` to null, and possibly put out a Debug warning that this occurs?

**Comment by Ekagra** [comment]
@david-allison-1 I can't clearly understand you, could you kindly explain it again?

**Comment by david-allison-1** [comment]
https://github.com/ankidroid/Anki-Android/blob/05c58c2f02d10ba2a8a338af427e12c3c7d6926d/AnkiDroid/src/main/java/com/ichi2/themes/StyledProgressDialog.java#L61-L70

If we never want a `""` to go through, we can add.

```java
if ("".equals(title)) {
    title = null;
    Timber.w("Invalid title was provided. Using null");
}
```

This fixes the issue permanently for users, and notifies developers that code changes are required.

**Comment by Ekagra** [comment]
got it should I implement this instead?

**Comment by david-allison-1** [comment]
> got it should I implement this instead?

I feel we might as well do both

**Comment by Ekagra** [comment]
alright should make a commit changing `StyledProgressDialog` now ?

**Comment by david-allison-1** [comment]
yep

**Comment by Ekagra** [comment]
added a  commit kindly review it @david-allison-1

**Review by david-allison-1 — APPROVED** [review]
Thanks! Looks good to me!

**Comment by david-allison-1** [comment]
Just to note:

Could you try and keep the commit messages to 50 characters in the future?

Commit messages can have an arbitrarily sized body: https://chris.beams.io/posts/git-commit/

We can squash/merge and change it for now, but it helps keep our commit log readable

**Comment by Ekagra** [comment]
yeah sure noted!

File: AnkiDroid/src/main/java/com/ichi2/themes/StyledProgressDialog.java [hunk 1]
```diff
@@ -60,6 +60,10 @@ public static MaterialDialog show(Context context, CharSequence title, CharSeque
 
     public static MaterialDialog show(Context context, CharSequence title, CharSequence message,
             boolean cancelable, DialogInterface.OnCancelListener cancelListener) {
+        if (title.equals("")) {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/themes/StyledProgressDialog.java:63]: If title is null, then this fails

```suggestion
        if ("".equals(title)) {
```

Comment (Ekagra) [reply AnkiDroid/src/main/java/com/ichi2/themes/StyledProgressDialog.java]: fixed

**Review by david-allison-1 — APPROVED** [review]

File: AnkiDroid/src/main/java/com/ichi2/themes/StyledProgressDialog.java [hunk 2]
```diff
@@ -60,6 +60,10 @@ public static MaterialDialog show(Context context, CharSequence title, CharSeque
 
     public static MaterialDialog show(Context context, CharSequence title, CharSequence message,
             boolean cancelable, DialogInterface.OnCancelListener cancelListener) {
+        if ("".equals(title)) {
+            title = null;
+            Timber.w("Invalid title was provided. Using null");
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/themes/StyledProgressDialog.java:65]: A `Timber.w` is a warning, in our convention that means "something is definitely wrong, but we may recover and continue".

So, if "something is definitely wrong", how can the same commit be *adding* uses of the style that triggers a warning?

It seems to me that there is nothing wrong with using null and we don't even need an informative message, we are *expecting* to send null and handling it now. At most this should be `Timber.d` now, just so developers aren't surprised but that's it I think
```suggestion
            Timber.d("Invalid title was provided. Using null");
```

**Review by mikehardy — CHANGES_REQUESTED** [review]
I think it should be Timber.d not Timber.w as mentioned

File: AnkiDroid/src/main/java/com/ichi2/themes/StyledProgressDialog.java (hunk 2 above)
Comment (Ekagra) [reply AnkiDroid/src/main/java/com/ichi2/themes/StyledProgressDialog.java]: done

Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/themes/StyledProgressDialog.java]: The commit is removing `""`, the source of the warning, not adding it.

But `.d` works - we'll pick it up in logcat on a debug machine regardless

Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/themes/StyledProgressDialog.java]: Ah, sorry was thinking opposite @david-allison-1 - I still don't think it warrants a .w though, .d will note it for devs as mentioned

**Review by mikehardy — APPROVED** [review]

**Comment by mikehardy** [comment]
Just pending CI (and labeled as such) anyone can squash-merge when/if they see CI pass

**Comment by Ekagra** [comment]
how to get this PR merged now ?

**Comment by mikehardy** [comment]
Yep - merged, just needed CI to pass and it just did :rocket:

**Comment by Ekagra** [comment]
glad to have my first PR merge 🥳

**Comment by mikehardy** [comment]
Hi there!

Just a friendly notice (one per person for PRs merged in March, regardless of PR quantity) that we try to process OpenCollective payments monthly - it's time for March 2021 submissions

If you are interested in compensation for this work, the process with details is here:

https://github.com/ankidroid/Anki-Android/wiki/OpenCollective-Payment-Process#how-to-get-paid

Thanks!

**Pull request merged**